xtable-core/src/main/java/org/apache/xtable/parquet/ParquetStatsExtractor.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.parquet.column.Encoding;
+import org.apache.parquet.column.statistics.Statistics;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+import org.apache.xtable.model.storage.FileFormat;
+import org.apache.xtable.model.config.InputPartitionFields;
+import org.apache.hadoop.conf.Configuration;
+
+@Value
+@Builder
+public class ParquetStatsExtractor {
+
+    private static final ParquetStatsExtractor INSTANCE = null; // new ParquetStatsExtractor();
+    @Builder.Default
+    private static final ParquetPartitionValueExtractor partitionExtractor =
+            ParquetPartitionValueExtractor.getInstance();
+    @Builder.Default
+    private static final ParquetSchemaExtractor schemaExtractor =
+            ParquetSchemaExtractor.getInstance();
+    @Builder.Default
+    private static final ParquetMetadataExtractor parquetMetadataExtractor =
+            ParquetMetadataExtractor.getInstance();
+
+    private final InputPartitionFields partitions;
+
+    public static ParquetStatsExtractor getInstance() {
+        return INSTANCE;
+    }
+
+    public static List<ColumnStat> getColumnStatsForaFile(ParquetMetadata footer) {
+        return getStatsForaFile(footer).values().stream()
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+    }
+    private Optional<Long> getMaxFromColumnStats(List<ColumnStat> columnStats) {
+        return columnStats.stream()
+                .filter(entry -> entry.getField().getParentPath() == null)
+                .map(ColumnStat::getNumValues)
+                .filter(numValues -> numValues > 0)
+                .max(Long::compareTo);
+    }
+
+
+    public static Map<ColumnDescriptor, List<ColumnStat>> getStatsForaFile(ParquetMetadata footer) {
+        List<ColumnStat> colStat = new ArrayList<ColumnStat>();
+        Map<ColumnDescriptor, List<ColumnStat>> columnDescStats = new HashMap<>();
+        for (BlockMetaData blockMetaData : footer.getBlocks()) {
+            MessageType schema = parquetMetadataExtractor.getSchema(footer);
+            List<ColumnChunkMetaData> columns = blockMetaData.getColumns();
+            columnDescStats =
+                    columns
+                            .stream()
+                            .collect(Collectors.groupingBy(columnMetaData -> schema.getColumnDescription(columnMetaData.getPath().toArray()),
+                                    Collectors.mapping(columnMetaData ->ColumnStat.builder()
+                                            .numValues(columnMetaData.getValueCount())
+                                            .totalSize(columnMetaData.getTotalSize())
+                                            .range(Range.vector(columnMetaData.getStatistics().genericGetMin(), columnMetaData.getStatistics().genericGetMax()))
```

Comment:
Inside of the [Range javadocs](https://github.com/apache/incubator-xtable/blob/main/xtable-api/src/main/java/org/apache/xtable/model/stat/Range.java#L44) there are some expectations for the value's type. Let's make sure there are tests asserting these values are being populating according to these expectations.